services/diddoc/queries/resources/resource_query_type_handler.go

```diff
@@ -0,0 +1,34 @@
+package resources
+
+import (
+	"github.com/cheqd/did-resolver/services"
+	"github.com/cheqd/did-resolver/services/diddoc/queries"
+	"github.com/cheqd/did-resolver/types"
+)
+
+type ResourceTypeHandler struct {
+	queries.BaseQueryHandler
+	ResourceHelperHandler
+}
+
+func (d *ResourceTypeHandler) Handle(c services.ResolverContext, service services.RequestServiceI, response types.ResolutionResultI) (types.ResolutionResultI, error) {
+	resourceType := service.GetQueryParam(types.ResourceType)
+	if resourceType == "" {
+		return d.Continue(c, service, response)
+	}
+
+	resourceCollection, err := d.CastToContent(service, response)
+	if err != nil {
+		return nil, err
+	}
+	// Filter the list of metadatas by the resourceCollectionId
```

Comment:
The noun `metadata` is uncountable. The plural form of metadata is also metadata.